src/chocolatey/infrastructure.app/commands/ChocolateyHelpCommand.cs

```diff
@@ -0,0 +1,303 @@
+        }
+
+        public virtual void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
+        {
+            configuration.Input = string.Join(" ", unparsedArguments);
+        }
+
+        public virtual void handle_validation(ChocolateyConfiguration configuration)
+        {
+        }
+
+        public virtual void help_message(ChocolateyConfiguration configuration)
+        {
+            this.Log().Info(ChocolateyLoggers.Important, "Help Command");
+            this.Log().Info(@"
+Print out the help command... Why are you asking for help on the help command?
+");
+
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Usage");
+            "chocolatey".Log().Info(@"
+    choco help
+");
+
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Examples");
+            "chocolatey".Log().Info(@"
+    choco help
+
+");
+      
+            "chocolatey".Log().Info(ChocolateyLoggers.Important, "Exit Codes");
+            "chocolatey".Log().Info(@"
+Exit codes that normally result from running this command.
+
+Normal:
+ - 0: operation was successful, no issues detected
+ - -1 or 1: an error has occurred
+
+Enhanced:
+ - 0: operation was successful, no issues detected
+ - -1 or 1: an error has occurred
+ - 2: no results (enhanced)
+
+NOTE: Starting in v0.10.12, if you have the feature '{0}' 
+ turned on, then choco will provide enhanced exit codes that allow 
+ better integration and scripting.
+
+If you find other exit codes that we have not yet documented, please 
+ file a ticket so we can document it at 
+ https://github.com/chocolatey/choco/issues/new/choose.
+
+".format_with(ApplicationParameters.Features.UseEnhancedExitCodes));
+        }
+
+        public virtual void noop(ChocolateyConfiguration configuration)
+        {
+            _packageService.list_noop(configuration);
+        }
+
+        public virtual void run(ChocolateyConfiguration configuration)
+        {
+            display_help_message();
+        }
+
+        public virtual IEnumerable<PackageResult> list(ChocolateyConfiguration configuration)
+        {
+            configuration.QuietOutput = true;
+            // here it's up to the caller to enumerate the results
+            return _packageService.list_run(configuration);
+        }
+
+        public virtual int count(ChocolateyConfiguration config)
+        {
+            config.QuietOutput = true;
+            return _packageService.count_run(config);
+        }
+
+        public virtual bool may_require_admin_access()
+        {
+            return false;
+        }
+        public static void display_help_message(Container container = null)
+        {
+                var commandsLog = new StringBuilder();
+                if (null == container)
+                {
+                    container = new Container();
+                }
+                IEnumerable<ICommand> commands = container.GetAllInstances<ICommand>();

```

Comment:
Unfortunately a new Container (when null) won't have any of these. So it might be worth it to state if the container is null to call `choco -?` to get a listing of all commands.

Reply:
🤔 yeah, it seems when calling `choco help` it's not coming with the container and thus isn't actually helpful. Not sure if the way I've updated it now makes sense, but it does resolve the container being null.